### PR TITLE
DOC-6681: Document FTS FLEX index

### PR DESCRIPTION
**Pull request by simon-dew**
The following draft documentation is ready for review:

* [SELECT Syntax › USE Clause › index-ref](https://simon-dew.github.io/docs-site/DOC-6681/server/6.6/n1ql/n1ql-language-reference/select-syntax.html#index-ref) — index name optional
* [USE Clause › USE INDEX Clause](https://simon-dew.github.io/docs-site/DOC-6681/server/6.6/n1ql/n1ql-language-reference/hints.html#use-index-clause) — index name optional
* [Full-Text Search Indexes in N1QL](https://simon-dew.github.io/docs-site/DOC-6681/server/6.6/n1ql/n1ql-language-reference/flex-indexes.html) — new page on N1FTY FLEX
* [Search Service › Accessing the Search Service with N1QL](https://simon-dew.github.io/docs-site/DOC-6681/server/6.6/learn/services-and-indexes/services/search-service.html#search_via_query) — add link to new N1FTY FLEX page

Docs issue: [DOC-6681](https://issues.couchbase.com/browse/DOC-6681)

### CODE REVIEW

File: modules/n1ql/examples/n1ql-language-reference/n1ql.ebnf [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+unnest-type ::= 'INNER' | ( 'LEFT' 'OUTER'? )
+
+/* USE Clause */
+
+use-clause ::= use-keys-clause | use-index-clause
+use-keys-clause ::= 'USE' use-keys-term
+use-keys-term ::= 'PRIMARY'? 'KEYS' expr
+use-index-clause ::= 'USE' use-index-term
+use-index-term ::= 'INDEX' '(' index-ref ( ',' index-ref )* ')'
+index-ref ::= index-type | index-name index-type?
```
Comment (sitaramv) [inline modules/n1ql/examples/n1ql-language-reference/n1ql.ebnf]: USE INDEX ( USING GSI)
USE INDEX ( ix1 USING GSI)
both valid, index-name is is optional

**Review by sitaramv — APPROVED** [review]

File: modules/n1ql/examples/n1ql-language-reference/n1ql.ebnf (hunk 1 above)
Comment (simon-dew) [reply modules/n1ql/examples/n1ql-language-reference/n1ql.ebnf]: sure, but `USE INDEX ( ix1 )` is valid as well (defaults to `USING GSI`)

Comment (simon-dew) [reply modules/n1ql/examples/n1ql-language-reference/n1ql.ebnf]: `USE INDEX ()` doesn't actually generate an error... 🤷‍♂️ but it seems meaningless

**Comment by binhquocle** [comment]
The Overview -> Restrictions section regarding pushdown could be misinterpreted as something that the user cannot do. So while it is good that we call these out, we should make it clear that this is only for performance consideration. Functionally, Limit/Offset/Agg will still work.

**Comment by simon-dew** [comment]
> The Overview -> Restrictions section regarding pushdown could be misinterpreted as something that the user cannot do. So while it is good that we call these out, we should make it clear that this is only for performance consideration. Functionally, Limit/Offset/Agg will still work.

Done

File: modules/n1ql/examples/n1ql-language-reference/n1ql.ebnf (hunk 1 above)
Comment (simon-dew) [reply modules/n1ql/examples/n1ql-language-reference/n1ql.ebnf]: Corrected the syntax to reflect this

File: modules/n1ql/pages/n1ql-language-reference/hints.adoc [hunk 2]
```diff
@@ -227,6 +224,23 @@ WHERE sourceairport = "SFO";
 ----
 ====
 
+.Use any suitable Full Text Search index
+====
+Specify that the query service should prefer an FTS index, without specifying the index by name.
+To qualify for this query, there must be an FTS index on state, city, and type, using the keyword analyzer.
```
Comment (abhinavdangeti) [inline modules/n1ql/pages/n1ql-language-reference/hints.adoc]: The FTS index doesn't need to index "city" for this case.
Optionally if it was a type-mapped index on "hotel", it doesn't need to have "type" indexed as well.

**Review by abhinavdangeti — COMMENTED** [review]
Looking good @simon-dew

File: modules/n1ql/pages/n1ql-language-reference/hints.adoc (hunk 2 above)
Comment (simon-dew) [reply modules/n1ql/pages/n1ql-language-reference/hints.adoc]: Done

**Pull request merged**